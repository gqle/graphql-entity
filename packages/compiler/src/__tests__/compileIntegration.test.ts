import { getCompilationFiles } from './getCompilationFiles'

const _defaultSchema = /* GraphQL */ `
  type Query {
    _isExtended: boolean
  }
`

describe('compiler', () => {
  const subject = async (source: string) => {
    return getCompilationFiles([['/entity.ts', source]])
  }

  test('basic regression test', async () => {
    const entitySource = /* GraphQL */ `
      type User {
        name: String!
      }

      extend type Query {
        getUser: User!
      }
    `

    const { files } = await subject(entitySource)

    for (const file of files) {
      expect(file.filepath.path).toMatchSnapshot('file path')
      expect(file.write()).toMatchSnapshot('file contents')
    }
  })

  test('root extensions', async () => {
    const entitySource = /* GraphQL */ `
      extend type Query {
        getString: String!
      }

      extend type Mutation {
        setString: String!
      }

      extend type Subscription {
        onString: String!
      }
    `

    const { files } = await subject(entitySource)

    for (const file of files) {
      expect(file.filepath.path).toMatchSnapshot('file path')
      expect(file.write()).toMatchSnapshot('file contents')
    }
  })
})
