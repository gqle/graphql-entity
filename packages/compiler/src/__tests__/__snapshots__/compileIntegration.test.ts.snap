// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler basic regression test: file contents 1`] = `
"// @gqle/generated
import { Maybe } from \\"@gqle/graphql-entity/dist/prelude\\";


/*
 * Definitions
 */

export interface User {
  name: string;
}


/*
 * Entities
 */

export type UserEntity = User


"
`;

exports[`compiler basic regression test: file contents 2`] = `
"// @gqle/generated
import { Awaitable, Maybe } from \\"@gqle/graphql-entity/dist/prelude\\";
import { createEntityServer as baseCreateEntityServer } from \\"@gqle/graphql-entity\\";
import { User as User } from \\"entity\\";


/*
 * Entities
 */

export interface Entities {
  User: User;
}


/*
 * Query
 */

export interface Query {
  getUser(): Awaitable<User>;
}


/*
 * Aliases
 */

export type QueryEntity = Query


/*
 * Server
 */

export const createEntityServer = (opts: { Query: QueryEntity }) =>
  baseCreateEntityServer<QueryEntity>(opts);

"
`;

exports[`compiler basic regression test: file path 1`] = `"/entity.ts"`;

exports[`compiler basic regression test: file path 2`] = `"/"`;
